### ISSUE TITLE
Add 2D embedding visualization (UMAP / t-SNE)

### ISSUE THREAD

**Issue by Khanz9664** [Khanz9664/TrustLens#7]
# [Feature] Add 2D Embedding Visualization (UMAP / t-SNE)

###  Problem Statement
The `embedding_separability` metric provides numerical scores, but "seeing is believing." Practitioners need a 2D projection of their high-dimensional embeddings to observe class clusters, overlaps, and manifold geometry visually.

###  Proposed Solution
Implement `plot_embedding_2d()` in the representation module. This function should project high-dimensional embeddings into 2D space and color-code them by class label.

###  Key Requirements
- **Algorithm Support**: Default to **UMAP** for preserving local/global structure, with **t-SNE** or **PCA** as a fallback.
- **Interactive Legend**: Clearly label each class cluster.
- **Metrics Overlay**: Annotate the plot with the Silhouette Score computed during analysis.
- **Sample Control**: If `n_samples` > 5000, implement automatic sub-sampling to keep runtime and plot density manageable.
- **Optional Dependencies**: Handle `umap-learn` or `scikit-learn` imports gracefully (don't force standard users to install them).

###  Expected Usage
```python
report.plot_embedding_2d(method="umap", save_path="clusters.png")

**Comment by WeiGuang-2099** [comment]
Hey, I just finished working on #25  and I'm looking for something more challenging to sink my teeth into. This feature caught my eye — 2D embedding visualization with algorithm fallback and sub-sampling logic seems like a great next step. Happy to discuss the approach before starting if that helps. Can I be assigned?

**Comment by Khanz9664** [comment]
Hey @WeiGuang-2099 ! Great to see you again 🙌

Your work on the binning logic earlier was excellent, so this is a perfect next step — assigning it to you 👍

This feature is quite impactful (visualization = usability), so here’s a clear implementation direction to keep things clean and consistent:

---

##  Function Design

Add `plot_embedding_2d()` to:
`trustlens/visualization/representation_plots.py`

```python
def plot_embedding_2d(
    embeddings: np.ndarray,
    labels: np.ndarray,
    silhouette_score: float | None = None,
    method: str = "umap",
    n_max: int = 5000,
    save_path: str | None = None,
    show: bool = True,
) -> plt.Figure:
```

---

##  Projection Logic (with fallback)

We want graceful handling of optional dependencies:

* `"umap"` → use `umap-learn` if available (preferred)
* `"tsne"` → fallback to `sklearn.manifold.TSNE`
* `"pca"` → final fallback (`sklearn.decomposition.PCA`)

👉 Use `try/except` imports inside the function so we don’t force extra dependencies on users.

---

##  Subsampling (important)

If `len(embeddings) > n_max`:

* reduce to `n_max` samples before projection
* simple random sampling is fine (stratified is optional)
* print/log a small message so users know subsampling occurred

---

##  Plot Design

* Use `matplotlib` scatter plot
* Color by class labels (e.g., `tab10`)
* Add legend
* Keep styling consistent with other TrustLens plots

---

##  Metrics Annotation

If `silhouette_score` is provided:

* display it in the title or as a small annotation

---

##  Expected Behavior

* Return `matplotlib.figure.Figure`
* Respect `show=False` (do not call `plt.show()`)
* Support `save_path` if provided

---

##  Integration (later step)

Once this is ready, we’ll wire it into `TrustReport.plot()`.

---

##  Before PR

```bash
pre-commit run --all-files
pytest
```

---

If you want to share a quick draft or approach before full implementation, feel free — happy to review early 👍

Looking forward to this, it’s a great feature 🚀

**Comment by WeiGuang-2099** [comment]
Thanks for the detailed write-up and the kind words! The implementation direction is very clear. I'll get started and aim to have a draft ready in a few days — will share it here before opening the PR.

**Comment by Khanz9664** [comment]
@WeiGuang-2099 Sounds great 👍

Feel free to share a draft anytime — happy to take a look early if helpful.

Looking forward to it 🚀
